src/HealthGPS.Console/model_parser.cpp

```diff
@@ -59,14 +59,35 @@ hgps::RiskFactorSexAgeTable load_risk_factor_expected(const host::Configuration
     return table;
 }
 
+std::unique_ptr<hgps::DummyModelDefinition>
+load_dummy_model_definition(hgps::RiskFactorModelType type, const poco::json &opt) {
+    MEASURE_FUNCTION();
+
+    // Get dummy model parameters
+    std::vector<hgps::core::Identifier> names;
+    std::vector<double> values;
+    std::vector<double> policy;
+    for (const auto &[key, json_params] : opt["ModelParameters"].items()) {
+        names.emplace_back(key);
+        values.emplace_back(json_params["Value"].get<double>());
+        policy.emplace_back(json_params["Policy"].get<double>());
+    }
+
+    return std::make_unique<hgps::DummyModelDefinition>(type, std::move(names), std::move(values),
+                                                        std::move(policy));
+}
+
 std::unique_ptr<hgps::RiskFactorModelDefinition>
 load_static_risk_model_definition(const std::string &model_name, const poco::json &opt,
                                   const host::Configuration &config) {
     // Load this static model with the appropriate loader.
-    if (hgps::core::case_insensitive::equals(model_name, "hlm")) {
+    if (model_name == "dummy") {
```

Comment:
Why on Earth is C++ happy to make this comparison but unable to use a `switch` statement with strings? It does not make any sense!!

Reply:
https://xkcd.com/353/